src/compiler/checker.ts

```diff
@@ -22226,11 +22228,6 @@ namespace ts {
                     }
 
                     switch (location.kind) {
-                        case SyntaxKind.SourceFile:

```

Comment:
This comes from the second `Update` comment -- exported symbols aren't really the same as the symbols in scope.